dspace/bin/dspace

```diff
@@ -35,13 +35,14 @@ if [ "$1" = "classpath" ]; then
   exit 0
 fi
 
-
-
 #Allow user to specify java options through JAVA_OPTS variable
 if [ "$JAVA_OPTS" = "" ]; then
   #Default Java to use 256MB of memory
   JAVA_OPTS="-Xmx256m -Dfile.encoding=UTF-8"
 fi
 
 # Now invoke Java
-java $JAVA_OPTS -classpath $FULLPATH org.dspace.app.launcher.ScriptLauncher "$@"
+java $JAVA_OPTS \
+     -classpath $FULLPATH \
+     -Dlog4j.configurationFile=$DSPACEDIR/config/log4j2.xml \
```

Comment:
Is there a way to not "hardcode" this into our commandline scripts, and simply read it from `log.init.config` setting in our dspace.cfg?

Reply:
Maybe just leave it unconfigured and let basic logging appear on the console?

Reply:
I fear that if we try to use the DSpace configuration to initialize logging, which is needed by the DSpace configuration code, we'll wind up again in a strange loop that tries to reconfigure logging after a bunch of it has already happened.  Something as simple as create-administrator in v6 pours out a long screed of Hibernate's muttering before settling down to work.

There are good reasons for logging to be entirely separate from the logged code.

Reply:
Removing the `-D` altogether sounds fine to me (if that works).  I just wasn't sure we wanted a path hardcoded in these scripts.